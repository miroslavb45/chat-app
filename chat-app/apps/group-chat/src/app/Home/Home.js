import React from 'react';
import Topbar from '../Topbar';

function Home() {
  return (
    <div>
      <Topbar />
    </div>
  );
}

export default Home;
