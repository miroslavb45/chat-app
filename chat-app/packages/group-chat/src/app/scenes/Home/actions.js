import { createAction } from '@reduxjs/toolkit';
import createRestAction from '../../../utils/restActionCreator';
import createWsAction from '../../../utils/wsActionCreator';

export const selectChannelAction = createAction('SELECT_CHANNEL_INTERNAL');
export const selectPrivateMessageAction = createAction('SELECT_PRIVATE_MESSAGE_INTERNAL');
export const unselectMainContentAction = createAction('UNSELECT_MAIN_CONTENT_INTERNAL')


export const { action: getJoinedChannels, success: getJoinedChannelsSuccess, error: getJoinedChannelsError } = createRestAction('GET_JOINED_CHANNELS', 'joinedChannels', 'GET' );