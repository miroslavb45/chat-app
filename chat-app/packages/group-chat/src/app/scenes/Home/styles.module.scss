@import '~stylesheets/abstracts';

.componentsWrapper{
  display: flex;
}
.wrapper {
  height: 100vh;
  background-size: cover;


  //////////////////

  .chatContainer {
    $top: 140px;
    position: fixed;
    right: 0;
    top: $top;
    width: 345px;
    height: calc(100% - #{$top});
    border-radius: 4px 0px 0px 0px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    backdrop-filter: blur(60px);
    background: rgba(40, 38, 39, 0.6);
    overflow: hidden;

    @supports not (backdrop-filter: blur(60px)) {
      background-color: #24252b;
    }

    //they say this is an ios specific property. we need this cuz on paper ios supports blur background but in reality it doesn't..
    @supports (-webkit-touch-callout: none) {
      background-color: #24252b;
    }

    @include media('>=lg+') {
      backdrop-filter: unset;
    }

    &.chat-enter {
      transform: translateX(345px);
      opacity: 0;

      @supports (-webkit-touch-callout: none) {
        opacity: 1;
      }
    }

    &.chat-enter-active {
      transform: translateX(0px);
      opacity: 1;
      transition: transform 400ms ease-in-out, opacity 400ms ease-in-out;
    }

    &.chat-exit {
      transform: translateX(0px);
      opacity: 1;
    }

    &.chat-exit-active {
      transform: translateX(345px);
      transition: transform 400ms ease-in-out, opacity 400ms ease-in-out;
      opacity: 0;
    }
  }

  .toggleChatButton {
    align-items: center;
    background: $lapalma;
    bottom: calc(50vh - 50px);
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.25);
    color: $white;
    cursor: pointer;
    height: 85px;
    justify-content: center;
    position: fixed;
    right: 0;
    text-transform: uppercase;
    width: 24px;
    font-size: 14px;
    line-height: 20px;
    border-radius: 10px 0px 0px 10px;

    &.isOpen {
      right: 345px;
      @include media('height<=xs+') {
        bottom: 0;
      }
    }

    @include media('>=sm-') {
      display: flex;
    }
    i {
      padding-right: 6px;
    }

    .toggleChatButtonLabel {
      display: flex;
      align-items: center;
      transform: rotate(-90deg);
    }

    &.button-enter {
      transform: translateX(345px);
    }

    &.button-enter-active {
      transform: translateX(0px);
      transition: transform 400ms ease-in-out;
    }

    &.button-exit {
      transform: translateX(-345px);
    }

    &.button-exit-active {
      transform: translateX(0px);
      transition: transform 400ms ease-in-out;
    }
  }
  //////////////
}
