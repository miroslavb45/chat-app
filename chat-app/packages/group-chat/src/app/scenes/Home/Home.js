import { getAuth } from 'firebase/auth';
import background from 'images/background.png';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { connectToWebsocket } from '../../shared/services/websocket';
import Topbar from '../../shared/Topbar';
import VideoChat from '../VideoChat/VideoChat';
import { getWorkspaceAction } from '../Workspace/actions';
import { getJoinedChannels } from './actions';
import { Channel } from './components/Channel';
import { CreateChannelModal } from './components/Channel/components/CreateChannelModal';
import DeleteChannelModal from './components/Channel/components/DeleteChannelModal/DeleteChannelModal';
import { RenameChannelModal } from './components/Channel/components/RenameChannelModal';
import { Messaging } from './components/Messaging';
import { getWorkspaceUsers } from './components/Messaging/actions';
import { DeleteMessagingModal } from './components/Messaging/components/DeleteMessagingModal';
import ProfileSettingsModal from './components/ProfileSettingsModal/ProfileSettingsModal';
import { Sidebar } from './components/Sidebar';
import styles from './styles.module.scss';

class Home extends Component {
  state = {
    isChatOpen: true,
    username: null,
    chatMessages: [],
    participants: [],
    pendingMessages: [],
    user: {},
    websocket: null,
    selectedUserEmail: null,
  };

  async componentDidMount() {
    const { dispatch } = this.props;
    const auth = getAuth();

    this.setState({ user: auth.currentUser });
    dispatch(getWorkspaceAction());
    dispatch(
      getWorkspaceUsers({
        entityId: this.props.workspaceId,
      })
    );

    await connectToWebsocket();

    dispatch(
      getJoinedChannels({
        entityId: this.props.workspaceId,
      })
    );
  }

  toggleChat = () => {
    this.setState({ isChatOpen: !this.state.isChatOpen });
  };

  handleUserCall = (e) => {
    this.setState({ selectedUserEmail: e });
  };

  get activeMainContent() {
    const { activeMainContent } = this.props;
    switch (activeMainContent?.type) {
      case 'channel':
        return <Channel id={activeMainContent.id} />;
      case 'message':
        return <Messaging id={activeMainContent.id} />;
      default:
        return null;
    }
  }
  render() {
    return (
      <>
        <div className={styles.wrapper} style={{ backgroundImage: `url(${background})` }}>
          <div className={styles.overlay}></div>
          <Topbar />
          <div className={styles.sidebarWrapper}>
            <Sidebar></Sidebar>
          </div>

          <div className={styles.mainWrapper}>{this.activeMainContent}</div>

          {this.props.isModalOpen && (
            <>
              <CreateChannelModal></CreateChannelModal>
              <RenameChannelModal></RenameChannelModal>
              <DeleteChannelModal></DeleteChannelModal>

              <DeleteMessagingModal></DeleteMessagingModal>
              <ProfileSettingsModal />
            </>
          )}
        </div>

        <VideoChat></VideoChat>
      </>
    );
  }
}

const mapStateToPros = (state) => ({
  activeMainContent: state.home.activeMainContent,
  workspaceId: state.workspace.activeWorkspace.id || state.workspace.activeWorkspace,
  isModalOpen: state.home.isModalOpen,
});

export default connect(mapStateToPros)(Home);
