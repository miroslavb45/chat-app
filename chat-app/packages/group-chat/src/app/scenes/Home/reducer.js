import { createReducer } from '@reduxjs/toolkit';

import { getWorkspacesSuccess } from './actions';

const initialState = {
  workspaces: [],
};

export default createReducer(initialState, {
  [getWorkspacesSuccess]: (state, { payload }) => {
    console.log(state, payload);

    state.workspaces = [...payload.workspaces];
  },
});
