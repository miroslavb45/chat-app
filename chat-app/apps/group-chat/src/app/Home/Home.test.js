import { render } from '@testing-library/react';
import React from 'react';
import Home from './Home';

describe('Home', () => {
    const defaultProps = {};

    it('should render', () => {
        const props = {...defaultProps};
        const { asFragment, queryByText } = render(<Home {...props} />);

        expect(asFragment()).toMatchSnapshot();
        expect(queryByText('Home')).toBeTruthy();
    });
});
