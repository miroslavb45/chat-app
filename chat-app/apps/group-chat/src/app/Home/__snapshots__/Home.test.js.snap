// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<DocumentFragment>
  Home
</DocumentFragment>
`;
